### ISSUE TITLE
[118, -12] Certificate Error on installing Moonlight (Tizen 8.0 TVs)

### ISSUE THREAD

**Issue by raghavkanwal** [OneLiberty/moonlight-chrome-tizen#37]
![image](https://github.com/OneLiberty/moonlight-chrome-tizen/assets/5167823/0bde75d8-b6a0-4f6f-9a3d-7c72488caa20)

TV model is QN65QN85DBFXZC - It's a 2022 model.

I've confirmed that develop mode is on, docker engine is running via Docker Desktop. I'm not sure what the solution to the root cause of this error is: 

```Transferring the package...
Transferred the package: /home/moonlight/Moonlight.wgt -> /home/owner/share/tmp/sdk_tools/tmp
Installing the package...
--------------------
Platform log view
--------------------
install MoonLightS.MoonlightWasm
package_path /home/owner/share/tmp/sdk_tools/tmp/Moonlight.wgt
app_id[MoonLightS.MoonlightWasm] install start
app_id[MoonLightS.MoonlightWasm] installing[9]
app_id[MoonLightS.MoonlightWasm] installing[19]
app_id[MoonLightS.MoonlightWasm] install failed[118, -12], reason: Check certificate error : :Invalid certificate chain with certificate in signature.:<-3>
spend time for wascmd is [1203]ms
Failed to install Tizen application.```

**Comment by OneLiberty** [comment]
Someone reported the same issue on the discord. 
It's kind of weird, i'll try to fix that.

I thought this was an issue related to the 2024 models. But it seems that i was wrong. 
I'll keep you informed !

**Comment by raghavkanwal** [comment]
@OneLiberty I'm sorry, I was incorrect, this is a 2024 model.

**Comment by Kingo64** [comment]
I ran into the same issue when upgrading from KyroFrCode's version. I uninstalled it in the apps list and restarted the TV but it seems like I didn't fully uninstall under the hood. Therefore, you might already have existing install of Moonlight even if it isn't showing up in the TV apps list. You can confirm this with:
`sdb shell 0 vd_applist | grep Moonlight`

If you can see results for `MoonLightS.MoonlightWasm` then you may need to uninstall it with `sdb shell 0 vd_appuninstall MoonLightS.MoonlightWasm`

Afterwards you can install as normal, this resolved the issue for me.

**Comment by raghavkanwal** [comment]
> I ran into the same issue when upgrading from KyroFrCode's version. I uninstalled it in the apps list and restarted the TV but it seems like I didn't fully uninstall under the hood. Therefore, you might already have existing install of Moonlight even if it isn't showing up in the TV apps list. You can confirm this with: `sdb shell 0 vd_applist | grep Moonlight`
> 
> If you can see results for `MoonLightS.MoonlightWasm` then you may need to uninstall it with `sdb shell 0 vd_appuninstall MoonLightS.MoonlightWasm`
> 
> Afterwards you can install as normal, this resolved the issue for me.

I just tried this just in case, but `sdb shell 0 vd_applist | grep Moonlight` returned nothing. It's a brand new 2024 TV that I just bought last week, so I doubt it already had moonlight installed earlier, haha. But no dice either way.

Still stuck on the same problem, any advice appreciated.

**Comment by jroselle** [comment]
Hello, just wanted to add I am seeing the same issue, 2024 S95D model TV.  Fresh install, first "developer mode" app so likely nothing else installed. 

app_id[MoonLightS.MoonlightWasm] install start
app_id[MoonLightS.MoonlightWasm] installing[9]
app_id[MoonLightS.MoonlightWasm] installing[19]
app_id[MoonLightS.MoonlightWasm] install failed[118, -12], reason: Check certificate error : :Invalid certificate chain with certificate in signature.:<-3>
spend time for wascmd is [875]ms

Thanks for any help provided, hopefully I won't need to sign up for a Samsung dev account to get the signatures...

**Comment by jroselle** [comment]
Wanted to followup for other folks on this - I did need to go install tizen studio to get a Samsung developer cert specifically and then sign the package with that cert.  So still open but right now probably not an easier way to do it?

**Comment by OneLiberty** [comment]
It seems like the only way until we can figure a way. 
yeah, i'll keep it open until there is a better way of installing it for 2024 models. 

For anyone reading this and having this issue : 

You need to install Tizen Studio and sign the app using your own Samsung Account
You can use this project [Moonlight.zip](https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip) to create the package.

**Comment by jhonpedro** [comment]
I'm sorry, but I'm not really sure of what to do once I have the Tizen Studio installed.

Can you please give more details?

**Comment by OneLiberty** [comment]
This is quite a long process. 
I can't explain everything. 
But you can follow this https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools 
follow from "Installing the tools" up to "Install the App" 

Few things worth mentioning :
- You only need TV extensions and certificate manager. No need to install anything else. 
- You might be able to sign the app with a Tizen certificate but i recommend using a Samsung certificate profile. 
- If you can't import the app, just create a new one named Moonlight (a TV app of course) and drag/drop all the files from the zip into the newly created project.

I don't have the time to create a full guide on how to do it but here are the main steps :

1. Install [Tizen Studio](https://developer.tizen.org/development/tizen-studio/download), [Certificate Manager and TV extensions](https://docs.tizen.org/application/dotnet/get-started/certificates/installing-the-extension/). 
2. [Import](https://developer.samsung.com/smarttv/develop/getting-started/creating-tv-applications/importing-tv-applications.html) the [App](https://github.com/user-attachments/files/18407152/Moonlight.zip)
 as a Tizen TV project. (Alternatively you can create a new project and copy/paste all the file from the archive into it.) 
3. Launch the [remote device manager](https://developer.samsung.com/smarttv/develop/getting-started/using-sdk/tv-device.html) and connect to the TV (this imply that you've activated developer mode).
4. [Create a certificate](https://developer.samsung.com/smarttv/develop/getting-started/setting-up-sdk/creating-certificates.html) using the certificate manager. (A Samsung one, which require an account). 
5. Launch/Run the app by right-clicking the project folder "Run As" -> "Tizen Web Application"

Voilà, hope this helps 

EDIT: Updated the link to the app for the latest version

**Comment by jhonpedro** [comment]
I followed this tutorial on the first link installing the extensions and the steps that you gave, and it totally worked!

Thank you!!

**Comment by maishuliang** [comment]
> This is quite a long process. I can't explain everything. But you can follow this https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools follow from "Installing the tools" up to "Install the App"
> 
> Few things worth mentioning :
> 
> * You only need TV extensions and certificate manager. No need to install anything else.
> * You might be able to sign the app with a Tizen certificate but i recommend using a Samsung certificate profile.
> * If you can't import the app, just create a new one named Moonlight (a TV app of course) and drag/drop all the files from the zip into the newly created project.
> 
> I don't have the time to create a full guide on how to do it but here are the main steps :
> 
> 1. Install [Tizen Studio](https://developer.tizen.org/development/tizen-studio/download), [Certificate Manager and TV extensions](https://docs.tizen.org/application/dotnet/get-started/certificates/installing-the-extension/).
> 2. [Import](https://developer.samsung.com/smarttv/develop/getting-started/creating-tv-applications/importing-tv-applications.html) the [App](https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip) as a Tizen TV project. (Alternatively you can create a new project and copy/paste all the file from the archive into it.)
> 3. Launch the [remote device manager](https://developer.samsung.com/smarttv/develop/getting-started/using-sdk/tv-device.html) and connect to the TV (this imply that you've activated developer mode).
> 4. [Create a certificate](https://developer.samsung.com/smarttv/develop/getting-started/setting-up-sdk/creating-certificates.html) using the certificate manager. (Preferably Samsung which require an account, but a Tizen might work).
> 5. Launch/Run the app by right-clicking the project folder "Run As" -> "Tizen Web Application"
> 
> Voilà, hope this helps

I just bought a new Samsung TV QN85 Series and I was facing this error while installing.
Some additional steps that took me some time to explore.

After step 4, I have to right-click on the project and select ```Build Signed Package``` before step 5 and it works for me.
I would suggest adding your post to README for newer models so people won't need to fail first and need to look around to find how to troubleshoot their issues.

Thanks for supporting this package btw.

**Comment by jhonpedro** [comment]
Yes, it would be good for new users to have a warning referencing this issue for newer Samsung models! 

Also, thank you again for your help!

**Comment by CanisLupus-ML** [comment]
I get this error when trying to install:
![tizen](https://github.com/user-attachments/assets/47c98576-d44f-4827-a9a0-7f26e17f1525)

Also installing via docker or USB doesn't work. Samsung TV the frame 2024. (QE75LS03DAUXXH).
Any help?

**Comment by alejus27** [comment]
> This is quite a long process. I can't explain everything. But you can follow this https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools follow from "Installing the tools" up to "Install the App"
> 
> Few things worth mentioning :
> 
> * You only need TV extensions and certificate manager. No need to install anything else.
> * You might be able to sign the app with a Tizen certificate but i recommend using a Samsung certificate profile.
> * If you can't import the app, just create a new one named Moonlight (a TV app of course) and drag/drop all the files from the zip into the newly created project.
> 
> I don't have the time to create a full guide on how to do it but here are the main steps :
> 
> 1. Install [Tizen Studio](https://developer.tizen.org/development/tizen-studio/download), [Certificate Manager and TV extensions](https://docs.tizen.org/application/dotnet/get-started/certificates/installing-the-extension/).
> 2. [Import](https://developer.samsung.com/smarttv/develop/getting-started/creating-tv-applications/importing-tv-applications.html) the [App](https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip) as a Tizen TV project. (Alternatively you can create a new project and copy/paste all the file from the archive into it.)
> 3. Launch the [remote device manager](https://developer.samsung.com/smarttv/develop/getting-started/using-sdk/tv-device.html) and connect to the TV (this imply that you've activated developer mode).
> 4. [Create a certificate](https://developer.samsung.com/smarttv/develop/getting-started/setting-up-sdk/creating-certificates.html) using the certificate manager. (Preferably Samsung which require an account, but a Tizen might work).
> 5. Launch/Run the app by right-clicking the project folder "Run As" -> "Tizen Web Application"
> 
> Voilà, hope this helps

Hi, I'm doing this same process as I have the cert issue, the .zip you provided has the most updated version so far?
Thx!

**Comment by OneLiberty** [comment]
@alejus27 Mmmh, i don't think it has the latest updates. 
I'll update it later in the month, when i get some free time. 

Edit: link updated to 1.2.1 latest commit (+fix for gamemode issue)

**Comment by superRFstrat** [comment]
I may be an idiot, but I can't get Tizen Studio to recognize the attached zip archive as a project.  I just set up an entirely new environment.  When using import, I can't get the project to populate into the list whether I point it at the archive, or if I extract the archive and point it at the root directory.

I have everything that's listed in the tizen-twitch article, and then some.  All of the tools I have are the 8.0 version, since it's a fresh install of the Tizen development tools.  I think that this issue may be related to the fact that I can only get the 8.0 versions of the tools, and not the 5.5 version that this was built on.  I haven't made anything using these tools, so I can very well be completely off base.

Any advice is very appreciated.  Thank you in advance!

![Screenshot of archive path in Import Dialog](https://github.com/user-attachments/assets/c0971be9-d96b-4e9f-8328-88529cf15c72)
![Screenshot of blank table when proceeding to the next page](https://github.com/user-attachments/assets/ad8257c0-50aa-44c2-bab5-36143fcacd76)

**Comment by OneLiberty** [comment]
@superRFstrat Just create a new project and copy paste the content of the zip in the new folder

No need to "import" anything 

Most of it is described here https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#issuecomment-2118982596

**Comment by DanielColwill** [comment]
> I may be an idiot, but I can't get Tizen Studio to recognize the attached zip archive as a project. I just set up an entirely new environment. When using import, I can't get the project to populate into the list whether I point it at the archive, or if I extract the archive and point it at the root directory.
> 
> I have everything that's listed in the tizen-twitch article, and then some. All of the tools I have are the 8.0 version, since it's a fresh install of the Tizen development tools. I think that this issue may be related to the fact that I can only get the 8.0 versions of the tools, and not the 5.5 version that this was built on. I haven't made anything using these tools, so I can very well be completely off base.
> 
> Any advice is very appreciated. Thank you in advance!
> 
> ![Screenshot of archive path in Import Dialog](https://private-user-images.githubusercontent.com/9374851/378231197-c0971be9-d96b-4e9f-8328-88529cf15c72.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1Nzk5NjMsIm5iZiI6MTcyOTU3OTY2MywicGF0aCI6Ii85Mzc0ODUxLzM3ODIzMTE5Ny1jMDk3MWJlOS1kOTZiLTRlOWYtODMyOC04ODUyOWNmMTVjNzIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjJUMDY0NzQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjJmMzVlNjI0YjQ4MmY5MzA4YzZkZDAzZGQzZTYxM2NjYzE3ZTcxZGYzM2E4MzliMmNkNWY5MTViNTBhMzUwOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.rgfIRg5108Tm2VSnkqIdio31OcQd8HKSstu9AKBZ1-A) ![Screenshot of blank table when proceeding to the next page](https://private-user-images.githubusercontent.com/9374851/378231210-ad8257c0-50aa-44c2-bab5-36143fcacd76.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1Nzk5NjMsIm5iZiI6MTcyOTU3OTY2MywicGF0aCI6Ii85Mzc0ODUxLzM3ODIzMTIxMC1hZDgyNTdjMC01MGFhLTQ0YzItYmFiNS0zNjE0M2ZjYWNkNzYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjJUMDY0NzQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDc5YThmMjA0MGQ5YzcwMTZkZDQ2Zjg2OGZjZDQ1OGM4ZTNhM2NhZThhZDVmMDZhMTYyY2Y0N2ZlNzg5ODA3NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.GeknllPctVKX3fSSkOaJeAzXqVet1sultsRWibmsgYc)

I can't get it to work either, there's a number of options when you select Tizen -> Template -> Web application, which I assume must be the right options as there's js files when extracting the folder.

**Comment by DanielColwill** [comment]
I can't get it to work either, there's a number of options when you select Tizen -> Template -> Web application, which I assume must be the right options as there's js files when extracting the folder.

**Comment by superRFstrat** [comment]
@OneLiberty I have tried creating new projects with several of the Tizen templates, and replacing the contents of the project with the files from the provided zip archive.  It has either complained about the config.xml or it doesn't recognize the build configuration.  The samsung-8.0 or tizen-8.0 will disappear from next to the project root in the explorer panel.  I got it to install via USB, so I'm not in a huge rush.  I'd just like to help provide insight to you, and possibly your team.  I'll be able to post screenshots and such later, if you'd like.

**Comment by jhonpedro** [comment]
I'll try to create a video for you guys with the way I did it. I did it 5 months ago, so I'll probably need to remember about the way I did it.

**Comment by superRFstrat** [comment]
@jhonpedro That'd be awesome!  Thanks in advance

**Comment by agusyoung** [comment]
@superRFstrat I just created this account to let you know how I fixed the config error; for me, it was something like "icon128.png does not exist". Sorry in advanced since I am typing this out from memory.
I just had to go to the moonlight zip file, I extracted it, opened config.xml with notepad, look for that "icon128.png" line of text and change it to "icon.png". Save. Then, do the same thing you were doing. Create the project with that new moonlight folder (in my case importing it did not work), just keep pressing next and before you try to "Run as", I right clicked the project and clicked "Build Signed Package", then you run it as "1 Tizen Web Application".
Hope that helps!

**Comment by DanielColwill** [comment]
hmm that seems to be on the right track @agusyoung, i can see two errors when trying to build project
` ./icons/icon.png : File does not exist.	config.xml	/tvmoonlight	line 8	XML Validation Problem`

`Attribute 'pointing-device-support' is not allowed to appear in element 'tizen:setting'.	config.xml	/tvmoonlight	line 20	XML Schema Validation Problem
`

**Comment by DanielColwill** [comment]
if you look at the log files when trying to run, it comes up with some java errors too
`DRIVE:\tizen-studio\platforms\tizen-8.0\tv-samsung\widlprocxmls\`

i dont see a tizen-studio\platforms folder

**Comment by agmalaga2020** [comment]
# Solution to Certificate Error on Tizen OS 8.0 for Developers

This tutorial will help you resolve the certificate error that appears when trying to install apps on Samsung TVs with Tizen OS 8.0 or higher, especially the `[118, -12] Certificate Error on installing` issue.

## Step 1: Create a Samsung Developer Account

1. **Register as a Developer on Samsung**:
   - If you don’t have a Samsung account yet, first register on the Samsung portal: [Samsung Account](https://account.samsung.com/).
   - Then, convert your account into a developer account by signing in on [Samsung Developers](https://developer.samsung.com/).

## Step 2: Install Samsung Extensions in Tizen Studio

1. **Open the Package Manager in Tizen Studio**:
   - Open Tizen Studio and select **Tools > Package Manager**.

2. **Install Samsung Certificate Extension**:
   - In the Package Manager, locate **Samsung Certificate Extension** and click **Install**.
   - Also, ensure that you have installed the **Samsung TV Extensions**, as they are necessary for working with TV apps.

## Step 3: Create a Samsung Certificate in Tizen Studio

1. **Open Certificate Manager**:
   - In Tizen Studio, go to **Tools > Certificate Manager**.
   - Click the **+** button to create a new certificate profile.

2. **Select Samsung Certificate**:
![Screenshot_38](https://github.com/user-attachments/assets/e09ee124-ba4e-4db3-b90a-f2c18cfda603)

   - Assign a name to the certificate profile and select **Samsung Certificate** instead of **Tizen Certificate**.
   - Sign in with your Samsung developer account when prompted by Tizen Studio.

4. **Complete the Certificate Form**:
   - Fill in the required fields (such as author name and password) to complete the certificate creation.

## Step 4: Sign the Package with the Samsung Certificate

1. Go back to your project in Tizen Studio.
2. Select the Samsung certificate profile you just created.
3. Build the signed package (**Build Signed Package**) to generate a `.wgt` file that you can install on your TV.

With these steps, you should be able to install your app on a Samsung TV without certificate issues.

---

This guide will help users resolve the specific certificate problem.

**Comment by CrazYoshi** [comment]
Just a quick remark. When creating the distributor certificate you need to enter your DUID from your TV.
I can be found at Menu > Support > Contact Samsung > Unique Device ID

**Comment by OneLiberty** [comment]
@CrazYoshi You're right ! 
But connecting to the TV before creating the certificate should automatically add the DUID to the certificate in the creation process. 

I should have mentioned it regardless of that. I'll update my post to make it clearer for the user !

**Comment by D-Wing0** [comment]
> # Solution to Certificate Error on Tizen OS 8.0 for Developers
> 
> This tutorial will help you resolve the certificate error that appears when trying to install apps on Samsung TVs with Tizen OS 8.0 or higher, especially the `[118, -12] Certificate Error on installing` issue.
> ## Step 1: Create a Samsung Developer Account
> 
>     1. **Register as a Developer on Samsung**:
>        
>        * If you don’t have a Samsung account yet, first register on the Samsung portal: [Samsung Account](https://account.samsung.com/).
>        * Then, convert your account into a developer account by signing in on [Samsung Developers](https://developer.samsung.com/).
> 
> 
> ## Step 2: Install Samsung Extensions in Tizen Studio
> 
>     1. **Open the Package Manager in Tizen Studio**:
>        
>        * Open Tizen Studio and select **Tools > Package Manager**.
> 
>     2. **Install Samsung Certificate Extension**:
>        
>        * In the Package Manager, locate **Samsung Certificate Extension** and click **Install**.
>        * Also, ensure that you have installed the **Samsung TV Extensions**, as they are necessary for working with TV apps.
> 
> 
> ## Step 3: Create a Samsung Certificate in Tizen Studio
> 
>     1. **Open Certificate Manager**:
>        
>        * In Tizen Studio, go to **Tools > Certificate Manager**.
>        * Click the **+** button to create a new certificate profile.
> 
>     2. **Select Samsung Certificate**:
>        ![Screenshot_38](https://private-user-images.githubusercontent.com/71594667/382489844-e09ee124-ba4e-4db3-b90a-f2c18cfda603.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMDM1MjMsIm5iZiI6MTczMjAwMzIyMywicGF0aCI6Ii83MTU5NDY2Ny8zODI0ODk4NDQtZTA5ZWUxMjQtYmE0ZS00ZGIzLWI5MGEtZjJjMThjZmRhNjAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE5VDA4MDAyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlY2E3MjEzMTVhYmYxZmE5NjdlZWQ1YWY1MWQ1ZjIzNTM4OGRkNTEwNGViZGQzMzkxMmViY2YzZjg5YWViNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tsmIjoDmVVIldoOMz_8zmiBfzsV3BRFk64MzxpbHu30)
>        
>        * Assign a name to the certificate profile and select **Samsung Certificate** instead of **Tizen Certificate**.
>        * Sign in with your Samsung developer account when prompted by Tizen Studio.
> 
>     3. **Complete the Certificate Form**:
>        
>        * Fill in the required fields (such as author name and password) to complete the certificate creation.
> 
> 
> ## Step 4: Sign the Package with the Samsung Certificate
> 
>     1. Go back to your project in Tizen Studio.
> 
>     2. Select the Samsung certificate profile you just created.
> 
>     3. Build the signed package (**Build Signed Package**) to generate a `.wgt` file that you can install on your TV.
> 
> 
> With these steps, you should be able to install your app on a Samsung TV without certificate issues.
> 
> This guide will help users resolve the specific certificate problem.

I tried it but it still doesn't work.  I still get the error message "[118, -12] Certificate Error on installing" I have a 2024 QLED Smart TV and I'm trying to install Jellyfin. I already have a .wgt file that works on all other Smart TV. Since the update to 8.0 I can no longer install it on my TV.

**Comment by CrazYoshi** [comment]
> Just a quick remark. When creating the distributor certificate you need to enter your DUID from your TV. I can be found at Menu > Support > Contact Samsung > Unique Device ID

Did you also check my previous comment?

**Comment by D-Wing0** [comment]
> > Just a quick remark. When creating the distributor certificate you need to enter your DUID from your TV. I can be found at Menu > Support > Contact Samsung > Unique Device ID
> 
> Did you also check my previous comment?

Of course I did that. I have already done this successfully with other TV's. But since the update to 8.0 I always get the error [118, -12] Certificate Error on installing.

**Comment by Cosco2k** [comment]
> # Solution to Certificate Error on Tizen OS 8.0 for Developers
> 
> This tutorial will help you resolve the certificate error that appears when trying to install apps on Samsung TVs with Tizen OS 8.0 or higher, especially the `[118, -12] Certificate Error on installing` issue.
> ## Step 1: Create a Samsung Developer Account
> ...
> ## Step 4: Sign the Package with the Samsung Certificate
> 
>     1. Go back to your project in Tizen Studio.
> 
>     2. Select the Samsung certificate profile you just created.
> 
>     3. Build the signed package (**Build Signed Package**) to generate a `.wgt` file that you can install on your TV.

Hey, at first thanks a lot for your work! Im also new here, so please be kind if i do something the wrong way ;)

Is there any way to get a "full list" of things to to? This tutorial for me feels not complete. I have the same problems as [superRFstrat] has. I did all steps from 1 - 3 and this sems to work. But what then? "1. Go back to your project in Tizen Studio." There is no project? I loaded the ZIP (but can only find moonlightUSB.zip - is it the right one?) and tryed to open it but it does not work.

thanks for your help! :)

**Comment by D-Wing0** [comment]
I already have a finished wgt file but despite having a developer account it rejects the certificate even with the correct DUID

**Comment by agmalaga2020** [comment]
> Lo probé pero sigue sin funcionar. Sigo recibiendo el mensaje de error "[118, -12] Error de certificado al instalar". Tengo un Smart TV QLED 2024 y estoy intentando instalar Jellyfin. Ya tengo un archivo .wgt que funciona en todos los demás Smart TV. Desde la actualización a 8.0 ya no puedo instalarlo en mi TV.


It worked for me on a 2024 QLED Smart TV.
I followed the steps shared in this section by OneLiberty and  I used the provided zip file titled "Moonlight.zip." https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip

In my case, I just needed to obtain the developer certificate, install the Samsung Certificate Extension, and finally create the certificate in Tizen Studio by linking it to my Samsung developer account.

I did this on a new project, and it worked on a 2024 QLED Smart TV with Tizen OS 8.0.

**Comment by agmalaga2020** [comment]
> Hey, at first thanks a lot for your work! Im also new here, so please be kind if i do something the wrong way ;)
> 
> Is there any way to get a "full list" of things to to? This tutorial for me feels not complete. I have the same problems as [superRFstrat] has. I did all steps from 1 - 3 and this sems to work. But what then? "1. Go back to your project in Tizen Studio." There is no project? I loaded the ZIP (but can only find moonlightUSB.zip - is it the right one?) and tryed to open it but it does not work.
> 
> thanks for your help! :)

Hi!

I first followed the steps from https://github.com/OneLiberty/moonlight-chrome-tizen, but I encountered issues with the certificate and ended up here.

I created a new project and followed OneLiberty's steps along with the file they uploaded to this forum.

What I shared is just to emphasize that, in my case, it was necessary to register as a developer and then link this account in Tizen Studio.

YW

**Comment by Cosco2k** [comment]
> Hi!
> 
> I first followed the steps from https://github.com/OneLiberty/moonlight-chrome-tizen, but I encountered issues with the certificate and ended up here.
> 
> I created a new project and followed OneLiberty's steps along with the file they uploaded to this forum.

Thanks for your replay. But.. ^^
 - Where did you create a new project? In Docker oder in Tizen Studio? Tizen, right?
 - i have to use this file you linked above? https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip Just for the future: Where can i find this file at the website? At https://github.com/OneLiberty/moonlight-chrome-tizen i can only find the releases of the USB version? S
 
So what I understand so far is instead of using Docker, I can load the project into Tizen Studio, add the cerificate (steps 1-4 posted above) and then install moonlight with Tizen Studio?

**Comment by agmalaga2020** [comment]
> Thanks for your replay. But.. ^^
> 
> * Where did you create a new project? In Docker oder in Tizen Studio? Tizen, right?
> * i have to use this file you linked above? https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip Just for the future: Where can i find this file at the website? At https://github.com/OneLiberty/moonlight-chrome-tizen i can only find the releases of the USB version? S
> 
> So what I understand so far is instead of using Docker, I can load the project into Tizen Studio, add the cerificate (steps 1-4 posted above) and then install moonlight with Tizen Studio?


### **Hi Cosco2k**

---

**Where did I create the project?**  
I created the project in **Tizen Studio**, as I don’t know how to add a certificate from Docker. There may be a way to do it, but I’m not familiar with it.

**Do I need to use the Moonlight.zip file?**  
Yes, you need to use the [[Moonlight.zip](https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip)](https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip) file. This file contains the necessary elements to configure the project in Tizen Studio.  
**Note:** This is not the same as the USB releases available in the main repository's release tab.

**Helpful link for setup:**  
I used the link shared by **OneLiberty**, which explains how to install tools in Tizen Studio. You can check it out here: [[Tizen Package Manager: Installing the tools](https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools)](https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools).

---

### **Step-by-Step Process for Using Tizen Studio**

1. **Download the [Moonlight.zip](https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip) file** and extract it.  
2. **Open Tizen Studio and create a new web application project for TV** (make sure you have completed the steps outlined in **[How to view installed packages in Tizen Studio](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#:~:text=complete%20the%20installation.-,How%20to%20View%20Installed%20Packages%20in%20Tizen%20Studio,-1.%20Open%20the)** below).  
3. **Copy the files** from the zip into the newly created project. If you encounter issues with icons, adjust the `config.xml` (e.g., replace "icon128.png" with "icon.png").  
4. **Generate a Samsung certificate from the Certificate Manager in Tizen Studio** (this was the step I missed and explained in [my first message](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#:~:text=3%20weeks%20ago-,Solution%20to%20Certificate%20Error%20on%20Tizen%20OS%208.0%20for%20Developers,higher%2C%20especially%20the%20%5B118%2C%20%2D12%5D%20Certificate%20Error%20on%20installing%20issue.,-Step%201%3A%20Create)).  
5. **Sign the package** with the certificate (use the **Build Signed Package** option).  
6. **Install the application** by [connecting to the TV in developer mode](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#:~:text=or%20use%20OpenJDK.-,Enable%20Developer%20Mode%20on%20Your%20TV%3A,-Go%20to%20the).

---

**Confirmation:**  
This process, along with the shared link, helped me resolve certificate issues and successfully complete the installation.

---

### **How to View Installed Packages in Tizen Studio**

**1. Open the Package Manager:**  
   - Open **Tizen Studio**.  
   - In the top toolbar, click **Tools > Package Manager**.  
   - This will open the **Tizen Package Manager**, where you can manage your packages.

**2. View Installed Packages:**  
   - In the **Package Manager** window, select the **Installed Packages** tab.  
   - Here, you will see all the installed packages. These are the packages I currently have installed:

#### **Main SDK:**
   - **8.0 Tizen**
   - **Tizen SDK Tools**
   - **Deprecated**

#### **Extension SDK:**
   - **TV Extension 8.0**
   - **Samsung Certificate Extension**
   - **Samsung Wearable Extension**
   - **TV Extension Tools**

**3. Additional Details:**  
   - If you need to check specific versions or additional details, click the information icon **(i)** next to the corresponding package.

---

### **Important Note: Tizen Studio Prerequisites**
To ensure Tizen Studio works correctly, make sure to meet the following **prerequisites**:

1. **Compatible Operating System:**
   - Windows 10 or later (64-bit).
   - macOS Mojave 10.14.4 or later.
   - Linux Ubuntu 16.04 or later.

2. **Java Development Kit (JDK):**
   - You need **Java JDK 8** (mandatory). Download it from [[Oracle](https://www.oracle.com/java/technologies/javase-jdk8-downloads.html)](https://www.oracle.com/java/technologies/javase-jdk8-downloads.html) or use OpenJDK.

3. **Enable Developer Mode on Your TV:**
   - Go to the **Apps** panel on your Smart TV.  
   - Enter `12345` using the remote.  
   - Enable developer mode, input your PC's IP, and restart the TV.

4. **Install Necessary Extensions in Tizen Studio:**
   - Install the following from the Package Manager:
     - **TV Extension Tools**
     - **Samsung Certificate Extension**

---

I hope this updated guide, now including your installed packages, helps you complete the installation successfully. If you need further assistance, let me know! 😊

**Comment by Cosco2k** [comment]
finally!!! Thanks a lot for your help :)

**Comment by fontanon** [comment]
Thanks @agmalaga2020 for your guidance. I got this working thanks to it. 

I'm just leaving a few more notes after my experience just in case it helps others:

**Generate a Samsung certificate from the Certificate Manager in Tizen Studio**
- Get in advance your TV DUID (Device Unique ID). This can be tricky. In my case was under TV Menu Assistance->TV Info. You will see multiple ids and even maybe a "Device ID" which is not the proper one. Look for Device Unique ID.
- Go through the process of generate the Samsung certificate from the Certificate Manager in Tizen Studio and in the last step (Distributor Certificate) there is a list to include your DUID. IMPORTANT: if you do not specify your TV DUID there then the `[118, -12] Certificate Error` will persist
- Please know along the process of generating the samsung certificate form the Certificate Manager in Tizen Studio you will have to create a developer account in Samsung Developer  platform. Just to let you know.

**Build Signed Package**
- Once the Samsung certificate is generated and the Moonlight.zip extracted under the web app folder, you will find `Build Signed Package` with right click over the project tree. It was difficult for me to find. So that is why I'm telling here.
- Once the package is built you will find the `.wgt` file under your project directory.

**Transferring the package to the docker container**
- If you are using the Moonlight docker installation method, use `docker cp` command to copy the `.wgt` file inside the docker container and then run the `tizen install -n your-signed-package.wgt -t your-device-check-with-sdb-devices`.

**Comment by maiconsc** [comment]
First of all, thank you for sharing this fantastic project!

I've followed all the steps up until creating the `.wgt` file with my certificate, but I'm having trouble sending the file to my TV. **Could anyone provide more details on how to do that, please?** Specifically, how can I use `docker cp` command to copy the file to the TV?

I also tried using the Device Manager in Tizen Studio, but it won't open for some reason.

Any tips or suggestions would be greatly appreciated! Thanks in advance!

**Comment by maiconsc** [comment]
> First of all, thank you for sharing this fantastic project!
> 
> I've followed all the steps up until creating the `.wgt` file with my certificate, but I'm having trouble sending the file to my TV. **Could anyone provide more details on how to do that, please?** Specifically, how can I use `docker cp` command to copy the file to the TV?
> 
> I also tried using the Device Manager in Tizen Studio, but it won't open for some reason.
> 
> Any tips or suggestions would be greatly appreciated! Thanks in advance!

I wanted to share how I solved my issue, hoping it might help others in the community.

**Solution 1: Using Tizen Studio Device Manager**
1. Open Device Manager and connect your TV.
2. Right-click on the TV and select "Install app."
3. Send the `.wgt` file to your TV.

**Solution 2: Using Docker**

1. Run the following command to create the container: `docker run -it --rm ghcr.io/oneliberty/moonlight-chrome-tizen:samsung_wasm`
2. Find the container ID using `docker ps`
3. Copy the `.wgt` file from your PC to the container: `docker cp "<path_to_wgt>.wgt" <container_id>:/home/moonlight` (This command should be executed from a separate PowerShell or terminal window outside the container)
4. Return to the container, connect to your TV, and install the .wgt file using `tizen install -n <name_of_your_certificate_file>.wgt -t <tv_id>`

**Some Additional Tips**

**Finding the DUID**
To create a certificate, you’ll need your TV’s DUID. It's possible to use the Device Manager to find this: just right-click on your TV id and select "DUID" to view this information.

**Creating the the Tizen project**

1. New Project > Template > TV (TV-samsung v8.0) > Web Application > Empty
2. Extract the **contents** of the Moonlight **folder** of the `Moonlight.zip` into the web app folder of your project.
3. (Windows will asks to replace files. Click "OK.")
4. Build the Signed Package normally in Tizen Studio and send the generated ` .wgt` file to your TV.

I hope this helps others facing similar issues!

Thanks again to the contributors for this amazing project—keep up the great work!

**Comment by Okayler** [comment]
I needed to make a new Project and import all the files from the zip into it. After that i build signed project with a right click.
Then i run the web app. like described. Then it worked. When i only import the app via the import dialogue it doesnt worked .. also make sure to install also the samsung certification package without you dont get the certificate from samsung ..

**Comment by Wacheee** [comment]
In my case creating a new project it always failed when I tried to build it. 

It worked with the "import" option and it worked flawless.
First I downloaded the latest `Moonlight.wgt` from the realease section
[Moonlight.wgt](https://github.com/OneLiberty/moonlight-chrome-tizen/releases/download/samsung_wasm-12788366703/Moonlight.wgt)

Then File-> Import -> Tizen project -> Archive file option -> Moonlight.wgt -> Profile "tv-samgung"

There is an error in `config.xml` you have to change `<icon src="./icons/icon128.png"/>` to `<icon src="icon.png"/>`

Lastly i selected the project with right click -> Run as Tizen Web Application.

Hope it can help someone with problems.

@OneLiberty will there be a workaround in the near future to install it directly using the Docker image? Or is this the only fix available for now?

Thanks!

**Comment by OneLiberty** [comment]
@Wacheee For now it's the only way i've found to install it. I think you will always need to create a certificate. 
There are options to resign from cmd too. 
I've made a bad gui for sdb that is available on the discord for people to try on windows. 
It should work to install the app AFTER it has been resigned within tizen studio, so not really useful for tizen 8.0. (it was mainly meant for tizen 3 to 7)

**Comment by joanroig** [comment]
> It seems like the only way until we can figure a way. yeah, i'll keep it open until there is a better way of installing it for 2024 models.
> 
> For anyone reading this and having this issue :
> 
> You need to install Tizen Studio and sign the app using your own Samsung Account You can use this project [Moonlight.zip](https://github.com/OneLiberty/moonlight-chrome-tizen/files/15151178/Moonlight.zip) to create the package.

This was the only way to get it working with a 2024 model running Tizen 8.0. To use the latest version, that's what I did:
- Download the latest Moonlight.wgt
- Open Tizen Studio and import the project file, making sure the profile is "tv-samsung":
![Image](https://github.com/user-attachments/assets/f18ce43f-1a18-4c1e-8021-0c53b5aa9427)
- Add a Samsung certificate and set it as the active profile
- Connect to the TV and enable "Permit to install applications" (I needed to do this every time I connect to the TV):
![Image](https://github.com/user-attachments/assets/9502547a-e9d6-4d00-b630-00187984b282)
- Change the required_version in the config.xml to match the TV OS version
- Select the TV in the dropdown and then install the app with "Run As > Tizen Web Application"

Full config.xml:
```
<?xml version="1.0" encoding="UTF-8"?>
<widget xmlns="http://www.w3.org/ns/widgets" xmlns:tizen="http://tizen.org/ns/widgets" id="http://samsung.tv/MoonlightTizen" version="1.2.1" viewmodes="fullscreen">
    <access origin="*"></access>
    <tizen:application id="MoonLightT.MoonlightTizen" package="MoonLightT" required_version="8.0" launch_mode="caller"/>
    <content src="index.html"/>
    <description>Moonlight Game Streaming</description>
    <feature name="http://tizen.org/feature/screen.size.all"/>
    <icon src="icon.png"/>
    <name>Moonlight</name>
    <tizen:privilege name="http://tizen.org/privilege/unlimitedstorage"/>
    <tizen:privilege name="http://tizen.org/privilege/filesystem.write"/>
    <tizen:privilege name="http://tizen.org/privilege/tv.inputdevice"/>
    <tizen:privilege name="http://tizen.org/privilege/filesystem.read"/>
    <tizen:privilege name="http://tizen.org/privilege/internet"/>
    <tizen:privilege name="http://tizen.org/privilege/tv.audio"/>
    <tizen:privilege name="http://developer.samsung.com/privilege/productinfo"/>
    <tizen:profile name="tv-samsung"/>
    <tizen:setting screen-orientation="landscape" context-menu="enable" background-support="disable" encryption="disable" install-location="auto" hwkey-event="enable"/>
</widget>
```

**Comment by gerwant22** [comment]
> This is quite a long process. I can't explain everything. But you can follow this https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools follow from "Installing the tools" up to "Install the App"
> 
> Few things worth mentioning :
> 
> * You only need TV extensions and certificate manager. No need to install anything else.
> * You might be able to sign the app with a Tizen certificate but i recommend using a Samsung certificate profile.
> * If you can't import the app, just create a new one named Moonlight (a TV app of course) and drag/drop all the files from the zip into the newly created project.
> 
> I don't have the time to create a full guide on how to do it but here are the main steps :
> 
> 1. Install [Tizen Studio](https://developer.tizen.org/development/tizen-studio/download), [Certificate Manager and TV extensions](https://docs.tizen.org/application/dotnet/get-started/certificates/installing-the-extension/).
> 2. [Import](https://developer.samsung.com/smarttv/develop/getting-started/creating-tv-applications/importing-tv-applications.html) the [App](https://github.com/user-attachments/files/18407152/Moonlight.zip)
>    as a Tizen TV project. (Alternatively you can create a new project and copy/paste all the file from the archive into it.)
> 3. Launch the [remote device manager](https://developer.samsung.com/smarttv/develop/getting-started/using-sdk/tv-device.html) and connect to the TV (this imply that you've activated developer mode).
> 4. [Create a certificate](https://developer.samsung.com/smarttv/develop/getting-started/setting-up-sdk/creating-certificates.html) using the certificate manager. (A Samsung one, which require an account).
> 5. Launch/Run the app by right-clicking the project folder "Run As" -> "Tizen Web Application"
> 
> Voilà, hope this helps
> 
> EDIT: Updated the link to the app for the latest version

Can you update the link with the zip package?

**Comment by Jasonjys** [comment]
I was able to follow the guide and install moonlight on my S95D but I only see 30 and 60 fps options. I think the TV supports 120/144 refresh rate. Did I do anything wrong or is the moonlight version does not support 120 fps?
![Image](https://github.com/user-attachments/assets/79019ad2-0085-4b77-947d-9afd927b89d5)

**Comment by rkl92** [comment]
> I was able to follow the guide and install moonlight on my S95D but I only see 30 and 60 fps options. I think the TV supports 120/144 refresh rate. Did I do anything wrong or is the moonlight version does not support 120 fps? ![Image](https://github.com/user-attachments/assets/79019ad2-0085-4b77-947d-9afd927b89d5)

same here, also can't see decode times and many other settings are missing.

**Comment by MrPhaze62** [comment]
@Jasonjys This port has never made a 120hz/144hz option. There is another fork that does. You're free to check that out.

@raclariu I believe the .zip file may be a little behind in updates from the main docker method and stats are not implemented.

**Comment by Jasonjys** [comment]
@MrPhaze62 do you know which one I should check out? Really want that higher refresh rate.

**Comment by MrPhaze62** [comment]
> @MrPhaze62 do you know which one I should check out? Really want that higher refresh rate.

@Jasonjys The fork you're looking for is this one: 
https://github.com/toypoodlegaming/moonlight-chrome-tizen

While this has the higher refresh options, I have not personally tried this as my current monitor does not output 90hz or above, so I don't know how smooth it may be for you or if it is indeed outputting those refresh rates to the TV. Make sure you check their installation guide and any issues they may have pinned in their fork. If there is an issue, probably best to contact the dev of that fork.

**Comment by xfechx** [comment]
I followed this guide above, because USB version of Moonlight stopped working suddenly. I eventually managed to build Moonlight through Tizen Studio and then install the Moonlight app in the TV, but in my case - now the Moonlight app does not pass the trackpad mouse to Windows. In the usb version of Moonlight I had installed, it did pass the trackpad correctly. Does anyone know why is this and how can I make it work?

**Comment by MrPhaze62** [comment]
> I followed this guide above, because USB version of Moonlight stopped working suddenly. I eventually managed to build Moonlight through Tizen Studio and then install the Moonlight app in the TV, but in my case - now the Moonlight app does not pass the trackpad mouse to Windows. In the usb version of Moonlight I had installed, it did pass the trackpad correctly. Does anyone know why is this and how can I make it work?

@xfechx it could be possible that Tizen Studio may have stripped a few lines of code when building that may have cause mouse and keyboard issues at times. See this thread: [here](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/80)

*i could be wrong as i have not used tizen studio or own a tizen 8.0 tv,  So take my advice with a grain of salt here.*

**Comment by xfechx** [comment]
@MrPhaze62 thanks for your reply. I am confused if I have to download a later version of the app, or if to add the lines of code that get stripped by Tizen Studio.

**Comment by dx616b** [comment]
> This is quite a long process. I can't explain everything. But you can follow this https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools follow from "Installing the tools" up to "Install the App"
> 
> Few things worth mentioning :
> 
> * You only need TV extensions and certificate manager. No need to install anything else.
> * You might be able to sign the app with a Tizen certificate but i recommend using a Samsung certificate profile.
> * If you can't import the app, just create a new one named Moonlight (a TV app of course) and drag/drop all the files from the zip into the newly created project.
> 
> I don't have the time to create a full guide on how to do it but here are the main steps :
> 
> 1. Install [Tizen Studio](https://developer.tizen.org/development/tizen-studio/download), [Certificate Manager and TV extensions](https://docs.tizen.org/application/dotnet/get-started/certificates/installing-the-extension/).
> 2. [Import](https://developer.samsung.com/smarttv/develop/getting-started/creating-tv-applications/importing-tv-applications.html) the [App](https://github.com/user-attachments/files/18407152/Moonlight.zip)
>    as a Tizen TV project. (Alternatively you can create a new project and copy/paste all the file from the archive into it.)
> 3. Launch the [remote device manager](https://developer.samsung.com/smarttv/develop/getting-started/using-sdk/tv-device.html) and connect to the TV (this imply that you've activated developer mode).
> 4. [Create a certificate](https://developer.samsung.com/smarttv/develop/getting-started/setting-up-sdk/creating-certificates.html) using the certificate manager. (A Samsung one, which require an account).
> 5. Launch/Run the app by right-clicking the project folder "Run As" -> "Tizen Web Application"
> 
> Voilà, hope this helps
> 
> EDIT: Updated the link to the app for the latest version



hey thanks, this worked. btw where can i find the latest files from your last version ?

**Comment by void-mckenzie** [comment]
This worked great. To simplify the process, you can just download the .wgt file, import it as a project in Tizen Studio, build with the certificate, and use the "tizen install -n Moonlight.wgt -t QN55S90DAFXZA" command to install it. Replace the device id with your particular device. 


I tried to get the 144Hz support by using the other fork, but it's very obtuse to install and it doesn't work most of the time. I'm kindly requesting the maintainers to see if native 144Hz or 120Hz support is possible.

**Comment by MiracAkbulut** [comment]
hey guys. i installed succesfully thanks to here but i have a problem. Sound shuttering. I tried traditional every fucking way but its not helping.(i also using with another devices and in the other device there is nothing.) could this be related to this version????? i really need help

**Comment by MiracAkbulut** [comment]
> This worked great. To simplify the process, you can just download the .wgt file, import it as a project in Tizen Studio, build with the certificate, and use the "tizen install -n Moonlight.wgt -t QN55S90DAFXZA" command to install it. Replace the device id with your particular device.
> 
> I tried to get the 144Hz support by using the other fork, but it's very obtuse to install and it doesn't work most of the time. I'm kindly requesting the maintainers to see if native 144Hz or 120Hz support is possible.
hey @void-mckenzie can you be more spesific i need really help. where should i write that code after i build the project. i am little confused right now and i believe i need uptade the my moonlight but there is no file without that one in the #37.

**Comment by MiracAkbulut** [comment]
> Can you update the link with the zip package?


did you find the uptated zip package?

**Comment by MrPhaze62** [comment]
> hey guys. i installed succesfully thanks to here but i have a problem. Sound shuttering. I tried traditional every fucking way but its not helping.(i also using with another devices and in the other device there is nothing.) could this be related to this version????? i really need help

@MiracAkbulut unfortunately sound issues seems to be common with some samsung tvs. There's an issue on github about it. See [here](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/87)

**Comment by MiracAkbulut** [comment]
> [@MiracAkbulut](https://github.com/MiracAkbulut) unfortunately sound issues seems to be common with some samsung tvs. There's an issue on github about it. See [here](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/87)

:( i cant figure it out but i am almost sure it isn't about wifi because i am using gfn too wihout issue. also i cant have found the steam link for tizen os. So its looking bad. i guess i will buy a LOOOOOOONG hdmi.

**Comment by paniq** [comment]
i don't want to seem ungrateful but i spend a good two hours trying to get this ungodly docker image to run before learning that it doesn't even work for newer Tizen anymore. it's too late for me, the time has been spent, but it surely would help others to be warned of this right on the github page - and link straight to this ticket.

btw moonlight is my last hope since samsung discontinued support for steam link, so here's fingers crossed

**Comment by xfechx** [comment]
@paniq I successfully installed it on a OLED 55" (S85D) samsung. I cannot remember exactly which version, I think it was latest version. On which TV are you trying to install?

It runs good, only issues:
60FPS
Mouse not going through to host.

I tried a Moonlight USB before and it worked well (the Dualsense trackpad was sent as mouse to host machine) but it stopped working after some time, I think the permanent install is when you build it yourself and make a tizen studio/samsung account. I think I uploaded via Tizen brew (using my phone) rather than using Docker (I remember it was just very confusing to use the Docker instructions found around here).

If someone can point us to the right direction on how to build the latest Moonlight for Tizen with 120hz support and mouse support that'be great.

**Comment by ZEROX7** [comment]
i got the app onto my tv but its just a white screen wehn i open it, any toughts on that?

**Comment by luligodio** [comment]
Hi, I got the app, but whenever i try to put it ip mannualy it close the keyboard and i cant write anything. I also tried puting it in the other way, but the "continue" button dont do anything...

**Comment by superMoYoX** [comment]
Hi everyone,

I'm really sorry but I struggle with that.
Like the original poster, I got the certificate error error when trying to install the wgt of Moonlight via docker.

So I installed Tizen Studio, created a certificate (the Samsung one, with the Samsung account login process), imported the project as a Tizen 8.0 project (an TV 9.0 and tizen 9.0, I tried all).

By doing this, when I run the project as web app it just closes with the error "closed" (thanks that's helpful).

I tried with the docker cmd (certificate should be OK so...), same certificate error.

I even tried to push the certificate trough the device manager (right click - > permit app install) and to deploy the wgt trough it (right click - > install app) but nothing works 😞

Can somebody help a lost future Moonlight user?

Thanks! 

PS : I also tried the Build Signed Package found in[ this comment ](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#issuecomment-2369171056) with no success.

**Comment by MrPhaze62** [comment]
> Hi everyone,
> 
> I'm really sorry but I struggle with that.
> Like the original poster, I got the certificate error error when trying to install the wgt of Moonlight via docker.
> 
> So I installed Tizen Studio, created a certificate (the Samsung one, with the Samsung account login process), imported the project as a Tizen 8.0 project (an TV 9.0 and tizen 9.0, I tried all).
> 
> By doing this, when I run the project as web app it just closes with the error "closed" (thanks that's helpful).
> 
> I tried with the docker cmd (certificate should be OK so...), same certificate error.
> 
> I even tried to push the certificate trough the device manager (right click - > permit app install) and to deploy the wgt trough it (right click - > install app) but nothing works 😞
> 
> Can somebody help a lost future Moonlight user?
> 
> Thanks! 
> 
> PS : I also tried the Build Signed Package found in[ this comment ](https://github.com/OneLiberty/moonlight-chrome-tizen/issues/37#issuecomment-2369171056) with no success.

@superMoYoX it would seem tizen 9.0 has came out recently and is relatively new, so it's quite surprising that the tizen 8.0 certificate method does not work for tizen 9.0.

If you do manage to somehow get it to install, or if anyone else that has managed to get it to install on tizen 9.0, please update this thread (or make a new issue). I wish i could help out if i had a tizen 9.0 TV. Hopefully, someone else does come out and find a way.

I'd like to think that tizen studio is probably still a way to install it, but probably requires a different method or options to set it up compared to the tizen 8.0 certificate method, but i could be wrong. (I'm not a dev or an expert, so my words are worth a grain of salt here.)

**Comment by superMoYoX** [comment]
> [@superMoYoX](https://github.com/superMoYoX) it would seem tizen 9.0 has came out recently and is relatively new, so it's quite surprising that the tizen 8.0 certificate method does not work for tizen 9.0.
> 
> If you do manage to somehow get it to install, or if anyone else that has managed to get it to install on tizen 9.0, please update this thread (or make a new issue). I wish i could help out if i had a tizen 9.0 TV. Hopefully, someone else does come out and find a way.
> 
> I'd like to think that tizen studio is probably still a way to install it, but probably requires a different method or options to set it up compared to the tizen 8.0 certificate method, but i could be wrong. (I'm not a dev or an expert, so my words are worth a grain of salt here.)

Sorry, it's not clear enough 😅
I meant, I tried the three methods to import the project (as a Tizen vers 8, as a Tizen vers 9 and as a Samsung TV vers 9).
But my TV has s actually running Tizen 8.

**Comment by elzappo** [comment]
@superMoYoX 
I think I had the same issue. 
At least for me the problem was that with the release of Tizen 9 only the TV extension 9.0 is provided with the official repository. So for me the solution was to download the TV extension 8.0 repository [here](https://developer.samsung.com/smarttv/develop/tools/tv-extension/archive.html), add it as a local repository in the Package Manager (Configuration -> Extension SDK -> Add Repository -> Select the .zip file + Also deactivate the existing "Samsung Tizen TV SDK" or else the 8.0 won't appear in the extension SDK list) and then install the 8.0 extension.
After that, the other steps worked for me and hopefully for you too.

**Comment by superMoYoX** [comment]
> [@superMoYoX](https://github.com/superMoYoX) I think I had the same issue. At least for me the problem was that with the release of Tizen 9 only the TV extension 9.0 is provided with the official repository. So for me the solution was to download the TV extension 8.0 repository [here](https://developer.samsung.com/smarttv/develop/tools/tv-extension/archive.html), add it as a local repository in the Package Manager (Configuration -> Extension SDK -> Add Repository -> Select the .zip file + Also deactivate the existing "Samsung Tizen TV SDK" or else the 8.0 won't appear in the extension SDK list) and then install the 8.0 extension. After that, the other steps worked for me and hopefully for you too.

Hi ! Thanks for your answer !

So I did like you said : downloading the 8.0 TV extension from the website and add it via the Package Manager. So  now I can import the project as Tizen 8.0 project (it shows beside the project name). I erased and created again the certificate (a Samsung one, with the online authentication). 
But when I launch (Run as -> Tizen Web Application), it seems to build properly the project and then : _Failed, Reason : closed_ .

I tried (we never know ...) to install it back with the docker and the sdb connection, still has the certificate issue with this way.

I even tried to use the Device Manager (right-click on the TV, Install App) and picked the wgt, it seems doing nothing.

The "connection" apparently works, I can retrieve the installed apps from the TV for example, but that's all :(

**Comment by bussukr** [comment]
@superMoYoX 
Hello,
I also spent half a day trying to make it work, but no success...
Check this youtube video and manual in it's description: https://www.youtube.com/watch?v=FK74FQZK65s  
It works for me.

**Comment by superMoYoX** [comment]
> [@superMoYoX](https://github.com/superMoYoX) Hello, I also spent half a day trying to make it work, but no success... Check this youtube video and manual in it's description: https://www.youtube.com/watch?v=FK74FQZK65s It works for me.

Thanks ! Installation worked by following this tutorial.

**Comment by rafifos** [comment]
So, I found out something.

You don't really need to build Moonlight from source in order to install it on Tizen 8.0, you just have to sign the .wgt file.

1. Just follow the already documented steps (Install Tizen Studio, connect your TV, create a Certificate, allow the TV to install apps, etc)
2. Download the latest release from GitHub
3. Sign it with your certificate with `tizen package -t wgt -s Default -- "./Moonlight.wgt" -o ./signed` (`Default` is the name of the Certificate that I've created)
4. Install it to your TV with `tizen install -n ./signed/Moonlight.wgt -t QN55S90DAGXZD` (You can get the Target name with sdb list)

Works on my 55" S90D (FW 1301)

**Comment by brybalicious** [comment]
For reference, @ndriqimlahu has written up a detailed set of instructions for Tizen 8.0 users here: https://github.com/ndriqimlahu/moonlight-tizen/wiki/Installation-Guide#installation-using-tizen-studio

**Comment by brybalicious** [comment]
Also, important to note that when importing .wgt archives to Tizen Studio, often some of the configuration flags are stripped which has been confirmed, particularly for the mouse pointer as per https://github.com/OneLiberty/moonlight-chrome-tizen/issues/80

**Comment by TheFabbiusCorp** [comment]
Hi, I want to share my experience about this issue, and in general with this tool.

It took me a few hours to solve the certificate issue since I had to follow the whole process of installing Tizen Studio, creating a Samsung account, creating a Samsung certificate, importing the app into Tizen (yeah, that was a pain too), resigning, and eventually installing it via Docker. Once installed on my QE65QN90C, I tried to pair it with:
- Sunshine latest
- Sunshine 0.23.1
- Apollo latest
After trying for 2 days, nothing worked. I also enabled Firewall ports and UPnP (although I was connecting locally).
At last, I recalled having a Fire Stick and it took literally 3 minutes to make it work: looked for the app, downloaded, installed, and paired.

Thanks everyone for the instructions to fix the certificate issue though!

**Comment by tldmt** [comment]
Ugh, even after signing last built .wtg then trying to install it with `tizen install -n ./Moonlight.wgt -t <tv_device>` and checking if signature is valid and even checking if its really there, inside the damned .wtg; I'm still having this issue. 
Can't figure out what sorcery this Tizen Studio needs to deploy stuff or if it's some issue with this whimsical tv.
edit: nevermind, I managed it out. Now working, whimsically.

**Comment by Iluck19** [comment]
> This is quite a long process. I can't explain everything. But you can follow this https://github.com/fgl27/smarttv-twitch?tab=readme-ov-file#tizen-package-managerinstalling-the-tools follow from "Installing the tools" up to "Install the App"
> 
> Few things worth mentioning :
> 
> * You only need TV extensions and certificate manager. No need to install anything else.
> * You might be able to sign the app with a Tizen certificate but i recommend using a Samsung certificate profile.
> * If you can't import the app, just create a new one named Moonlight (a TV app of course) and drag/drop all the files from the zip into the newly created project.
> 
> I don't have the time to create a full guide on how to do it but here are the main steps :
> 
> 1. Install [Tizen Studio](https://developer.tizen.org/development/tizen-studio/download), [Certificate Manager and TV extensions](https://docs.tizen.org/application/dotnet/get-started/certificates/installing-the-extension/).
> 2. [Import](https://developer.samsung.com/smarttv/develop/getting-started/creating-tv-applications/importing-tv-applications.html) the [App](https://github.com/user-attachments/files/18407152/Moonlight.zip)
>    as a Tizen TV project. (Alternatively you can create a new project and copy/paste all the file from the archive into it.)
> 3. Launch the [remote device manager](https://developer.samsung.com/smarttv/develop/getting-started/using-sdk/tv-device.html) and connect to the TV (this imply that you've activated developer mode).
> 4. [Create a certificate](https://developer.samsung.com/smarttv/develop/getting-started/setting-up-sdk/creating-certificates.html) using the certificate manager. (A Samsung one, which require an account).
> 5. Launch/Run the app by right-clicking the project folder "Run As" -> "Tizen Web Application"
> 
> Voilà, hope this helps
> 
> EDIT: Updated the link to the app for the latest version

hello everyone! i had the same issue so i fix it with this method but now i have another issue.
when i try to connect my pc with sunshine my tv doesnt see my pc i try to connect manually but that doesnt work either

**Comment by shiipou** [comment]
Can be cool to install tools like this in the docker image : https://github.com/sreyemnayr/tizencertificates

**Comment by Aetf** [comment]
Can we use the approach in https://github.com/Georift/install-jellyfin-tizen to allow specifying a custom certificate with this docker image and resign the package with new certificate? I have already generated my certificate. If I can use that directly with this, it makes it possible to automate the installation.

**Comment by apeman76** [comment]
Use https://github.com/PatrickSt1991/Samsung-Jellyfin-Installer
It will handle all the certificate stuff for you. No need to make a certificate or really do anything other than going to settings to select a custom WGT (leave release and version empty) and click install. Really easy install this way.

**Comment by ghost** [comment]
this is some bullshit as i can not even install a proper app on a device i have payed 1700 euro's for... i am on linux so i can not use tizen studio. maybe this will work in a VM. 

i am at the breaking point of selling that piece of ewaste... 

@apeman76 at firsr ur sugestion did not worked and now for some reason it did. finaly, afther hours....

**Comment by e-dong** [comment]
FYI this can be automated via https://github.com/sreyemnayr/tizencertificates! 

This takes the pain away of creating samsung certificates.

@aamaanaa I am also on linux, you can install the tizen studio CLI
https://developer.tizen.org/development/tizen-studio/download/
`Tizen Studio 6.1 with CLI(command line interface) installer`

The download option says Ubuntu, but it's just a script. I'm running Artix (systemd less arch).

**Comment by Joopz0r** [comment]
> Use https://github.com/PatrickSt1991/Samsung-Jellyfin-Installer It will handle all the certificate stuff for you. No need to make a certificate or really do anything other than going to settings to select a custom WGT (leave release and version empty) and click install. Really easy install this way.

I downloaded this and the moonlight.wgt and installed was that easy. Thank you

**Comment by ghost** [comment]
> FYI this can be automated via https://github.com/sreyemnayr/tizencertificates!
> 
> This takes the pain away of creating samsung certificates.
> 
> [@aamaanaa](https://github.com/aamaanaa) I am also on linux, you can install the tizen studio CLI https://developer.tizen.org/development/tizen-studio/download/ `Tizen Studio 6.1 with CLI(command line interface) installer`
> 
> The download option says Ubuntu, but it's just a script. I'm running Artix (systemd less arch).

i treid using samsung jelly fin installer and then loaded custom wgt file...

still requires a samsung account, solved with the use of a temp mail, but i finally got it installed lol

**Comment by Tom60chat** [comment]
You can also use **TizenBrew Installer** (not the one to start TizenTube) by using `Install from GitHub` and put `OneLiberty/moonlight-chrome-tizen`, it got a self certsign included.

**Comment by caiammm** [comment]
This process is not friendly... I tried for many, many hours to set everything. And it's not clear in many, many steps... My next TV will be LG... I heard that it's easier to set up there. I'm very disappointed with Samsung
